### PR TITLE
Phase 2.1 allow users to auto-open app when users have the app installed

### PR DESCRIPTION
**Pull request by rubinsingh**
JIRA for Phase 2.1: https://branch.atlassian.net/browse/DEVEX-249

### CODE REVIEW

File: src/view/actions/autoOpenApp.html [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<html>
```
Comment (ethanneff) [inline src/view/actions/autoOpenApp.html]: add <!DOCTYPE html> to the top

File: src/view/actions/autoOpenApp.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+<head>
+    <title>Auto Open Your App</title>
+    <script src="https://assets.adobedtm.com/activation/reactor/extensionbridge/extensionbridge.min.js"></script>
+    <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+    <link rel="stylesheet" type="text/css" href="../resources/branch.css">
+    <link rel="stylesheet" type="text/css" href="../resources/autoOpenApp.css">
+    <script src="../resources/autoOpenAppViewHelper.js"></script>
```
Comment (ethanneff) [inline src/view/actions/autoOpenApp.html]: move js below body

File: src/view/actions/autoOpenApp.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
+<html>
+<head>
+    <title>Auto Open Your App</title>
+    <script src="https://assets.adobedtm.com/activation/reactor/extensionbridge/extensionbridge.min.js"></script>
```
Comment (ethanneff) [inline src/view/actions/autoOpenApp.html]: move js below body

File: src/view/actions/autoOpenApp.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                   spellcheck="false"
+                   class="branch-form-input"
+                   />
+            <div id="link-data">
+                <div class="link-data-rows">
+                    Deep Link Data:
+                    <div class="link-data-rows-hook"></div>
+                </div>
+                <div class="add-more-data">
+                    <button class="add-more-data-button" onclick="autoOpen.addRow();">+ More Data</button>
```
Comment (ethanneff) [inline src/view/actions/autoOpenApp.html]: you don't need a semicolon

File: src/view/actions/autoOpenApp.html [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    <script src="https://assets.adobedtm.com/activation/reactor/extensionbridge/extensionbridge.min.js"></script>
+    <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+    <link rel="stylesheet" type="text/css" href="../resources/branch.css">
+    <link rel="stylesheet" type="text/css" href="../resources/autoOpenApp.css">
+    <script src="../resources/autoOpenAppViewHelper.js"></script>
+</head>
+<body>
+<div class="auto-open-outer-shell">
+    <div>
```
Comment (ethanneff) [inline src/view/actions/autoOpenApp.html]: you can remove un-used <div>

File: src/view/actions/autoOpenApp.html [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            <input type="text"
+                   id="campaign"
+                   spellcheck="false"
+                   class="branch-form-input"
+                   />
+            <label for="tags">Tags:</label>
+            <input type="text"
+                   id="tags"
+                   spellcheck="false"
+                   class="branch-form-input"
```
Comment (ethanneff) [inline src/view/actions/autoOpenApp.html]: add placeholder="one, two, three"

File: src/view/resources/autoOpenAppViewHelper.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+autoOpen.removeRow = function(id) {
+	const child = autoOpen.getElementId(id.replace("-anchor", ""));
+	const parent = child.parentNode;
+	parent.removeChild(child);
+}
+
+autoOpen.getData = function() {
+	const channel = autoOpen.getElementId("channel").value;
+	const campaign = autoOpen.getElementId("campaign").value;
+	console.log(channel);
```
Comment (ethanneff) [inline src/view/resources/autoOpenAppViewHelper.js]: remove

File: src/view/resources/autoOpenAppViewHelper.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+	console.log(campaign);
```
Comment (ethanneff) [inline src/view/resources/autoOpenAppViewHelper.js]: remove

File: extension.json [hunk 9]
```diff
@@ -37,6 +37,29 @@
         "type": "object",
         "properties": {}
       }
+    },
+    {
+      "displayName": "Auto Open Your App",
+      "name": "autoopenapp",
+      "libPath": "src/lib/actions/autoOpen.js",
```
Comment (ethanneff) [inline extension.json:44]: should this action be renamed to linkCreator as well?

Comment (rubinsingh) [reply extension.json:44]: This is the action that will call .deepview() from the WebSDK so based on that, it has the correct name. The next phase will introduce a new action for sending SMSes which we will place in sms.js.

**Pull request merged**